pkg/vault/test/foundry/fork/BoostedPoolWithInitializedBuffer.t.sol

```diff
@@ -251,38 +258,134 @@ contract BoostedPoolWithInitializedBufferTest is BaseVaultTest {
 
         IBatchRouter.SwapPathExactAmountIn[] memory paths = _buildExactInPaths(
             tooLargeSwapAmount,
-            tooLargeSwapAmount / USDC_FACTOR - 1
+            tooLargeSwapAmount / USDC_FACTOR - 2
         );
 
-        snapStart("forkBoostedPoolSwapTooLarge-ExactIn");
         vm.prank(alice);
         (uint256[] memory pathAmountsOut, address[] memory tokensOut, uint256[] memory amountsOut) = batchRouter
             .swapExactIn(paths, MAX_UINT256, false, bytes(""));
-        snapEnd();
 
         _verifySwapResult(pathAmountsOut, tokensOut, amountsOut, vars);
     }
 
-    function testBoostedPoolSwapOutOfBufferRangeExactOut__Fork() public {
+    function testBoostedPoolSwapOutOfBufferRangeExactOut__Fork__Fuzz(uint256 tooLargeSwapAmount) public {
+        tooLargeSwapAmount = bound(
+            tooLargeSwapAmount,
+            2 * _MAX_SWAP_AMOUNT_WITHIN_BUFFER_RANGE,
+            _MAX_SWAP_AMOUNT_OUTSIDE_BUFFER_RANGE
+        );
+
         SwapResultLocals memory vars = _createSwapResultLocals(SwapKind.EXACT_OUT);
 
         IBatchRouter.SwapPathExactAmountOut[] memory paths = _buildExactOutPaths(
             tooLargeSwapAmount + DAI_TO_USDC_FACTOR,
             tooLargeSwapAmount / USDC_FACTOR
         );
 
-        snapStart("forkBoostedPoolSwapTooLarge-ExactOut");
+        // EXACT_IN DAI -> USDC does not introduce rounding issues, since the resulting amountOut is divided by 1e12
+        // to get the correct amount of USDC to return.
+        // However, EXACT_OUT DAI -> USDC has rounding issues, because amount out is given in 6 digits, and we want an
+        // 18 decimals amount in, which is not precisely calculated. The calculation below reproduces what happens in
+        // the vault to scale tokens in the swap operation of the boosted pool
```

Comment:
:100:

Reply:
I believe this comment was in there already, because of another test, but I agree it's a good comment, hahahaha